### PR TITLE
chore(trie): replace TrieAccount by alloy Account

### LINKED ISSUE #12318: trie: replace `TrieAccount` by alloy `Account`
### Describe the feature

At the moment, in the trie crate, we have a `TrieAccount` structure

https://github.com/paradigmxyz/reth/blob/d5f01036016263fbad6a3af9dc3707b2701adc1d/crates/trie/common/src/account.rs#L10-L21

which seems to contain the same information as its alloy counterpart:

https://github.com/alloy-rs/alloy/blob/f2f1001c9122703a35378dcdf20c5af42af2bc63/crates/consensus/src/account.rs#L5-L20

To avoid code duplication, it could be nice to remove the reth definition and directly import the alloy `Account` structure when needed around the reth codebase.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamuik16**
Closes https://github.com/paradigmxyz/reth/issues/12318

### CODE REVIEW

File: crates/trie/common/src/account.rs [hunk 1]
```diff
@@ -9,21 +9,32 @@ use revm_primitives::AccountInfo;
...
-    /// Account nonce.
-    pub nonce: u64,
-    /// Account balance.
-    pub balance: U256,
-    /// Account's storage root.
-    pub storage_root: B256,
-    /// Hash of the account's bytecode.
-    pub code_hash: B256,
-}
+pub struct TrieAccount(AlloyTrieAccount);
```
Comment (shekhirin) [inline crates/trie/common/src/account.rs:12]: can we instead use the Alloy struct directly?

Comment (kamuik16) [reply crates/trie/common/src/account.rs:12]: I tried this but there are some foreign impls implemented on this type which are violated when used Alloy struct instead.

**Comment by kamuik16** [comment]
Closing this for now :(

**Comment by mattsse** [comment]
oh @kamuik16 I think what we can do is move the from conversion to `alloy-genesis`

**Comment by kamuik16** [comment]
> oh @kamuik16 I think what we can do is move the from conversion to `alloy-genesis`

yep, I was talking about the from conversions only. We'll need to move it to alloy first and then we can directly replace the TrieAccount with the alloy one.

**Pull request closed without merging**